### ISSUE TITLE
question on horizon_lanes in Load_Data_new.py

### ISSUE THREAD

**Issue by yuqi1991** [wvangansbeke/LaneDetection_End2End#56]
Hi, 

about line 150
```python
horizon_lanes = []
        for lane in lanes:
            horizon_lanes.append(min([y_cord for (x_cord, y_cord) in zip(lane, h_samples) if x_cord != -2] or [self.resize]))
        y_val = min(horizon_lanes)
        horizon = torch.zeros(gt.size[1])
        horizon[0:int(np.floor(y_val))] = 1
```

I dont quite understand what  variable `horizon` is used for
Beside  `lane` and `h_samples` in `zip()` seem have different size (because previously `lane` has been extended to size 56 filled with -2, but `h_samples` doesn't), wouldn't it cause mismatched indexed coordination?

**Comment by wvangansbeke** [comment]
Hi @yuqi1991,

The horizon estimation is necessary to know the end of each lane line. The network just predicts the polynomials otherwise. 

Indeed , `lane` and `h_samples` need to have the same size. `h_samples` should also have 56 values in that particular case. The json test file from TuSimple contains 56 height values for each image. That's why I extended `lane`.

Best, Wouter

**Comment by yuqi1991** [comment]
@wvangansbeke Thanks for your reply.

In terms of the `h_samples` here ,I found it's read from the provided  `Label/lanes_ordered.json` file, in which it has only 48 values instead of 56. So to make it work properly, I should used the original `h_samples`( which starts from `160` instead of `240`) from TuSimple or extend it manually, am I correct ?

**Comment by wvangansbeke** [comment]
Hi @yuqi1991,

Yes, I that makes sense.

Both labels are computed relatively towards the position of the car. The position of the car is every time in the middle of the list, provided in the json files.
- `position`: how many lanes on the left and right side of the car? a `0` says there is no lane, a `1` says the opposite.
- `lines`: what's the type of line?  `-1` says no line, `0` says dotted line and a `1` says solid line. 

Best, 
Wouter
